### PR TITLE
Pynndescent

### PR DESCRIPTION
**Pull request by lmcinnes**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by lmcinnes** [comment]
Should be good to go now.

File: recipes/pynndescent/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: b58832a8f56d6d89018c5d2e9fce866ec7a5fa7b1a5aa02363d6ef2015929b31
+
+build:
+  number: 0
+  skip: True  # [py27 or win32]
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
```
Comment (xhochy) [inline recipes/pynndescent/meta.yaml]: No need for an empty build section
```suggestion
```

File: recipes/pynndescent/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: b58832a8f56d6d89018c5d2e9fce866ec7a5fa7b1a5aa02363d6ef2015929b31
+
+build:
+  number: 0
+  skip: True  # [py27 or win32]
```
Comment (xhochy) [inline recipes/pynndescent/meta.yaml]: I've taken a look at the sources and this is a pure Python package. Thus no need to build per Python version. The 3.6+ restriction can be solved in the `run` section
```suggestion
  noarch: python
```

File: recipes/pynndescent/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+  host:
+    - python
```
Comment (xhochy) [inline recipes/pynndescent/meta.yaml]:
```suggestion
    - python >=3.6
```

File: recipes/pynndescent/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - setuptools
+    - pip
+  run:
+    - python
```
Comment (xhochy) [inline recipes/pynndescent/meta.yaml]:
```suggestion
    - python >=3.6
```

File: recipes/pynndescent/meta.yaml (hunk 3 above)
Comment (lmcinnes) [reply recipes/pynndescent/meta.yaml]: Ah, thanks -- wasn't sure the best way to do this.

**Comment by lmcinnes** [comment]
Thanks for the feedback. One minor catch is that win32 isn't supported by numba, and so that case may have issues. Not sure how best to exclude it.

**Comment by xhochy** [comment]
win32 is automatically excluded by `numba` not being available there. `conda` will then correctly (and understandably) report to the user that it cannot install this package as its dependency numba is missing on win32.

**Pull request merged**